src/rust/receiver-mock/src/router/otlp.rs

```diff
@@ -101,18 +105,313 @@ fn otlp_anyvalue_to_string(anyvalue: &commonv1::AnyValue) -> String {
     return s;
 }
 
+pub async fn handler_receiver_otlp_metrics(
+    req: HttpRequest,
+    body: web::Bytes,
+    app_state: web::Data<AppState>,
+    opts: web::Data<options::Options>,
+) -> impl Responder {
+    let remote_address = get_address(&req);
+
+    let content_type = get_content_type(&req);
+
+    if let Some(response) = try_dropping_data(&opts, &content_type) {
+        return response;
+    }
+
+    match content_type.as_str() {
+        OTLP_PROTOBUF_FORMAT_CONTENT_TYPE => {
+            let metrics_data: metricsv1::MetricsData = match metricsv1::MetricsData::decode(&mut Cursor::new(body)) {
+                Ok(data) => data,
+                Err(_) => return HttpResponse::BadRequest().body("Unable to parse body"),
+            };
+            let mut result = MetricsHandleResult::new();
+
+            // TODO: Consider giving it some basic capacity to avoid too many allocations.
+            let mut lines = vec![];
+            for resource_metrics in metrics_data.resource_metrics {
+                let resource_attributes = &resource_metrics.resource.unwrap().attributes;
+                for instrumentation_lib_metrics in resource_metrics.instrumentation_library_metrics {
+                    for metric in instrumentation_lib_metrics.metrics {
+                        // Translating the metrics to prometheus format allows us to use the already existing API for parsing metrics into samples.
+                        let metric_sample_vec = sample::otlp_metric_to_samples(&metric, resource_attributes);
+
+                        for m in &metric_sample_vec {
+                            if opts.print.metrics {
+                                println!("metrics => {:?}", m);
```

Comment:
How does this look in practice?

Reply:
The comment is outdated if it's what you mean.

Reply:
```
metrics => Sample { metric: "otelcol_processor_groupbyattrs_log_groups_bucket", value: 4.0, labels: {"container": "otelcol", "instance": "10.244.0.23:8888", "le": "+Inf", "prometheus": "test-helm-default-ot-metadata-533/rel-1045518918
```

Reply:
Allright, I'm ok with this. What we should really do is define https://doc.rust-lang.org/std/fmt/trait.Display.html for `Sample` and use that everywhere we need to print metrics. Should do that in #293.